fix(builtins/diagnostics/todo_comments): convert filetype to lang before calling vim.treesitter.get_parser

thallada opened the pull request:
Fixes issue #1527. Same reproduction steps with this patch does not produce the syntax bug.

I also created a separate bug with nvim-treesitter about the syntax breaking when `get_parser` is called with an invalid lang: https://github.com/nvim-treesitter/nvim-treesitter/issues/4693

jose-elias-alvarez commented:
Thanks! Re: logging the error: the source is enabled for all filetypes by default, but most filetypes don't have Treesitter parsers (nor is the user guaranteed to have them all installed), so I'd support logging it at the `debug` or `trace` level so that it only shows up in logs. That would make it possible to debug issues like this without overloading users with notifications.

Also, I think we can get rid of [this line](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/b862270965aa616df4e78b01477e6007a6be4b72/lua/null-ls/builtins/diagnostics/todo_comments.lua#L93), since it should be handled by the correspondence table.

thallada commented:
@jose-elias-alvarez I added a debug log to the `get_parser` error case and removed the latex ft conversion line you mentioned (I also confirmed that `vim.treesitter.language.get_lang('tex')` does return 'latex').

jose-elias-alvarez commented:
LGTM, thanks for putting this together!
